components/esp32/gdbstub.c

```diff
@@ -258,11 +300,67 @@ static void sendReason() {
 	gdbPacketEnd();
 }
 
+//Fetch the task status
+static unsigned getAllTasksHandle(unsigned index, unsigned * handle, const char ** name, unsigned * coreId) {
+	static unsigned taskCount = 0;
+    static TaskStatus_t tasks[32];
+
+    if (!taskCount) {
+		unsigned runTime = 0;
+		taskCount = uxTaskGetNumberOfTasks();
+		taskCount = uxTaskGetSystemState(tasks, 32, &runTime);
```

Comment:
It's almost impossible to ensure the state of FreeRTOS task's chained list (unless validating each pointer in FreeRTOS but this would be a huge cost for a very sparsely used feature).

I wonder if it would be better to let it crash instead and:
1. Set a boolean to true the first time we are walking the FreeRTOS task list, 
2. If we re-enter the exception handler, check the boolean. If it's true, simply return the previous exception since it's saved from the first entry. It would then appear in gdb as a single (the running) task and the previous/initial exception frame.

What do you think ?

Reply:
I'm testing this idea right now, and I'll update the PR when I'm done with this. I've also added a conditional compilation of the FreeRTOS part of the gdbstub (based on the presence of `uxTaskGetSystemState`).